apps/framework-cli/src/infrastructure/olap/clickhouse/queries.rs

```diff
@@ -49,18 +49,23 @@ impl CreateTableQuery {
                 "Kafka('{}:{}', '{}', 'clickhouse-group', 'JSONEachRow') SETTINGS {}",
                 kafka_host, kafka_port, topic, KAFKA_SETTINGS,
             ),
+            true,
         )
     }
 
     pub fn build(
         table: ClickhouseTable,
         engine: String,
+        ignore_primary_key: bool,
```

Comment:
it looks like this depends on the engine - wouldn't it be better to have an if Kafka then that else this?

Feels like a weird parameter to ask the caller to set

Reply:
Yeah I don't really like that either, I changed that to an enum.